src/ConsumerConfig.cc

```diff
@@ -265,6 +291,39 @@ ConsumerConfig::ConsumerConfig(const Napi::Object &consumerConfig, pulsar_messag
     }
     pulsar_consumer_configuration_set_dlq_policy(this->cConsumerConfig.get(), &dlq_policy);
   }
+
+  if (consumerConfig.Has(CFG_BATCH_RECEIVE_POLICY) &&
+      consumerConfig.Get(CFG_BATCH_RECEIVE_POLICY).IsObject()) {
+    Napi::Object propObj = consumerConfig.Get(CFG_BATCH_RECEIVE_POLICY).ToObject();
+    int maxNumMessages = -1;
+    if (propObj.Has(CFG_BATCH_RECEIVE_POLICY_MAX_NUM_MESSAGES) &&
```

Comment:
I found Java client will use the `receiverQueueSize` if `maxNumMessages` is bigger than the `receiverQueueSize`, I'm not sure whether we can add this check in the node client.

Reply:
This logic has been added on the CPP client:  https://github.com/apache/pulsar-client-cpp/blob/main/lib/ConsumerImplBase.cc#L38-L49

Reply:
Got it, thanks.